helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java

```diff
@@ -267,6 +268,115 @@ public Response getClusterConfig(@PathParam("clusterId") String clusterId) {
     return JSONRepresentation(config.getRecord());
   }
 
+
+  @PUT
+  @Path("{clusterId}/customized-state-aggregation-config")
+  public Response addCustomizedStateAggregationConfig(@PathParam("clusterId") String clusterId,
+      String content) {
+    if (!doesClusterExist(clusterId)) {
+      return notFound(String.format("Cluster %s does not exist", clusterId));
+    }
+
+    HelixAdmin admin = getHelixAdmin();
+    ZNRecord record;
+    try {
+      record = toZNRecord(content);
+    } catch (IOException e) {
+      _logger.error("Failed to deserialize user's input " + content + ", Exception: " + e);
+      return badRequest("Input is not a vaild ZNRecord!");
+    }
+
+    try {
+      CustomizedStateAggregationConfig customizedStateAggregationConfig =
+          new CustomizedStateAggregationConfig.Builder(record).build();
+      admin.addCustomizedStateAggregationConfig(clusterId, customizedStateAggregationConfig);
+    } catch (Exception ex) {
```

Comment:
Can we not catch a more specific exception? It seems there could be runtime HelixException?

Reply:
Thanks for the comment. I still prefer to catch all the exception like we do in other functions.